pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -167,31 +174,147 @@ func checkVersion(instance *servingv1alpha1.KnativeServing) error {
+	if err := waitForServiceMeshMemberRoleReady(instance); err != nil {
+		return err
+	}
+	log.Info(fmt.Sprintf("Successfully configured %s namespace into configured members", instance.GetNamespace()))
+	return nil
+}
+
+// waitForServiceMeshControlPlaneReady checks whether serviceMeshControlPlane installs all required component
+func waitForServiceMeshControlPlaneReady() error {
+	smcp := &maistrav1.ServiceMeshControlPlane{}
+	if waitError := wait.PollImmediate(pollInterval, pollTimeout, func() (bool, error) {
+		if err := api.Get(context.TODO(), client.ObjectKey{Namespace: ingressNamespace, Name: "basic-install"}, smcp); err != nil {
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		for _, cond := range smcp.Status.Conditions {
+			if cond.Type == maistrav1.ConditionTypeReady && cond.Status == maistrav1.ConditionStatusTrue {
+				return true, nil
+			}
+		}
+		return false, nil
+	}); waitError != nil {
+		return waitError
+	}
+	return nil
+}
+
+// installServiceMeshControlPlane installs serviceMeshControlPlane
+func installServiceMeshControlPlane(instance *servingv1alpha1.KnativeServing) error {
+	const (
+		path = "deploy/resources/serviceMesh/smcp.yaml"
+	)
+	manifest, err := mf.NewManifest(path, false, api)
+	if err != nil {
+		log.Error(err, "Unable to create serviceMeshControlPlane install manifest")
+		return err
+	}
+	transforms := []mf.Transformer{mf.InjectOwner(instance)}
+	if len(ingressNamespace) > 0 {
+		transforms = append(transforms, mf.InjectNamespace(ingressNamespace))
+	}
+	if err := manifest.Transform(transforms...); err != nil {
+		log.Error(err, "Unable to transform serviceMeshControlPlane manifest")
 		return err
 	}
-	if err := api.Get(context.TODO(), client.ObjectKey{Namespace: "istio-system", Name: "default"}, smmr); err != nil {
-		return checkError(err, instance)
+	if err := manifest.ApplyAll(); err != nil {
+		log.Error(err, "Unable to install serviceMeshControlPlane")
+		return err
 	}
-	found := func(smmr *maistrav1.ServiceMeshMemberRoll) bool {
+	return nil
+}
+
+// installServiceMeshMemberRole installs serviceMeshMemberRole for knative-serving namespace
+func installServiceMeshMemberRole(instance *servingv1alpha1.KnativeServing) error {
+	smmr := &maistrav1.ServiceMeshMemberRoll{}
+	if err := api.Get(context.TODO(), client.ObjectKey{Namespace: ingressNamespace, Name: "default"}, smmr); err != nil {
+		if apierrors.IsNotFound(err) {
+			smmr.Name = "default"
+			smmr.Namespace = ingressNamespace
+			smmr.Spec.Members = []string{instance.GetNamespace()}
+			if err := api.Create(context.TODO(), smmr); err != nil {
+				return err
+			}
+			return nil
+		}
+		return err
+	} else {
+		var exist = false
+		// If serviceMeshMemberRole already exist than check for knative-serving ns is configured member or not
 		for _, member := range smmr.Status.ConfiguredMembers {
-			if member == "knative-serving" {
-				return true
+			if member == instance.GetNamespace() {
+				exist = true
```

Comment:
You can add a "break" statement here once you've found the namespace.

Reply:
This condition added to check whether existing `SMMR` have `knative-serving` ns or not if not update with `knative-serving`

If we add break statement then for all the cases `knative-serving` ns witll be updated in  SMMR so to avoid that added Boolean check

Reply:
Sure but once you set `exist` to true, you can `break` the outer for loop to not run unnecessary checks.